site/src/pages/RoadmapPage/AddYearPopup.tsx

```diff
@@ -16,99 +16,97 @@ const AddYearPopup: FC<AddYearPopupProps> = ({ placeholderName, placeholderYear
-      </Button>
-      <Overlay show={show} target={target} placement="top">
-        <Popover id="">
-          <Popover.Content>
-            <Form noValidate validated={validated} className="add-year-form">
-              <Form.Group>
-                <Form.Label className="add-year-form-label">Name</Form.Label>
-                <Form.Control
-                  required
-                  type="text"
-                  name="name"
-                  value={name}
-                  onChange={(e) => {
-                    setName(e.target.value);
-                  }}
-                  onKeyDown={(e: React.KeyboardEvent) => {
-                    // prevent submitting form (reloads the page)
-                    if (e.key === 'Enter') {
-                      e.preventDefault();
-                    }
-                  }}
-                  maxLength={35}
-                  placeholder={placeholderName}
-                ></Form.Control>
-              </Form.Group>
-              <Form.Group>
-                <Form.Label className="add-year-form-label">Start Year</Form.Label>
-                <Form.Control
-                  required
-                  type="number"
-                  name="year"
-                  value={year}
-                  onChange={(e) => {
-                    setYear(parseInt(e.target.value));
-                  }}
-                  onKeyDown={(e: React.KeyboardEvent) => {
-                    // prevent submitting form (reloads the page)
-                    if (e.key === 'Enter') {
-                      e.preventDefault();
-                    }
-                  }}
-                  min={1000}
-                  max={9999}
-                  placeholder={placeholderYear.toString()}
-                ></Form.Control>
-              </Form.Group>
-              <Button
-                className="popup-btn"
-                onClick={() => {
-                  if (name === '' || year < 1000 || year > 9999 || Number.isNaN(year)) {
-                    setValidated(true);
-                    return;
-                  }
-
-                  setValidated(false);
-                  setShow(!show);
-                  dispatch(
-                    addYear({
-                      yearData: {
-                        startYear: year,
-                        name: name.trim(),
-                        quarters: ['fall', 'winter', 'spring'].map((quarter) => {
-                          return { name: quarter, courses: [] };
-                        }),
-                      },
-                    }),
-                  );
-                  setYear(placeholderYear);
-                  setName(placeholderName);
-                }}
-              >
-                Add Year
-              </Button>
-            </Form>
-          </Popover.Content>
-        </Popover>
-      </Overlay>
+      <OverlayTrigger placement="top" overlay={overlay} trigger="click" rootClose>
```

Comment:
> This isn't completely true. I am using the onTrigger to call showQuarterMenu and update show because this is what allows clicking the buttons in the menu to actually close it. OverlayTrigger is handling the clicking outside and putting it into a onTrigger event for us and can keep the state on its own, but we want to also be able to have the state separate so we can control it manually for the case I just described.

I see. You are missing the onToggle and show attributes for the add year overlay trigger though.

Reply:
I'm assuming my commit just now is what you mean?

Reply:
Yes.